scipy/stats/_stats_py.py

```diff
@@ -501,6 +501,12 @@ def mode(a, axis=0, nan_policy='propagate', keepdims=False):
 
     """  # noqa: E501
     # `axis`, `nan_policy`, and `keepdims` are handled by `_axis_nan_policy`
+    if not np.issubdtype(a.dtype, np.number):
+        message = ("Argument `a` is not recognized as numeric. "
```

Comment:
Phrased like this to reduce confusion when the input appears to be numeric but isn't coerced to numeric type (e.g. `Series` with pandas integers).

Reply:
are you intending to drop `bool` on purpose here?

Reply:
Yes, for now. The deprecation warning was produced under the same conditions as the error message would be with this PR, so it should be OK.